var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -0,0 +1,93 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Amrex(CMakePackage):
+    """A Block-Structured AMR Framework. AMReX is the successor to BoxLib."""
+
+    homepage = "https://ccse.lbl.gov/AMReX/index.html"
+    url      = "https://github.com/AMReX-Codes/amrex.git"
+
+    version('develop', git='https://github.com/AMReX-Codes/amrex.git', tag='master')
+    version('17.06', git='https://github.com/AMReX-Codes/amrex.git', commit='836d3c7')
+
+    variant('mpi', default=True, description='Enable MPI parallel support')
+# variant('omp', default=False, description='Enable OpenMP parallel support')
```

Comment:
See answer below -- I think respecting the intended defaults is fine, so maybe set the *variant's* default to the same thing the build system does.  But when you add a (boolean) variant, it's either on or off, and if you don't explicitly set it and rely on defaults, it may end up being a lie.